src/client/gui/widget/checkbox_widget.cpp

```diff
@@ -0,0 +1,97 @@
+#include "checkbox_widget.h"
+
+#include "../component/rectangle_component.h"
+#include "../component/text_component.h"
+#include "widget_helper.h"
+
+namespace gui {
+
+    CheckBoxWidget::CheckBoxWidget(RectangleComponent* rectComponent, TextComponent* label)
+        : mp_rectangle(rectComponent)
+        , mp_label(label)
+        , isChecked(false)
+    {
+        componentList.push_back(mp_rectangle);
+        componentList.push_back(mp_label);
+
+        mp_label->setFontSize(40);
+    }
+
+    void CheckBoxWidget::setPosition(const GuiDimension& position)
+    {
+        mp_rectangle->setPosition(position);
+    }
+
+    void CheckBoxWidget::setSize(const GuiDimension& size)
+    {
+        mp_rectangle->setSize(size);
+    }
+
+    void CheckBoxWidget::setColour(float r, float g, float b)
+    {
+        mp_rectangle->colour = {r, g, b};
+    }
+
+    void CheckBoxWidget::setImage(int Image)
+    {
+        mp_rectangle->setTexture(Image);
+    }
+
+    void CheckBoxWidget::setText(const std::string& text)
+    {
+        mp_label->setText(text);
+    }
+
+    void CheckBoxWidget::setTextSize(unsigned size)
+    {
+        mp_label->setFontSize(size);
+    }
+
+    void CheckBoxWidget::handleClick(sf::Mouse::Button button, float mx, float my)
+    {

```

Comment:
Continuing on from above, this could toggle the checkbox like

```
isChecked = !isChecked;
```

This would toggle the bool between true and false. From this you could probably set the image of the checkbox by adding 2 properties to the checkbox, one for the `checked` image, and the other for the `unchecked` image. 

(if that makes sense 😛 )

eg from the Lua

```
checbox.checkedImage = checkboxTexture_checked 
checbox.unCheckedImage = checkboxTexture_checked 
```

This way the checkbox would have the behavior of a checkbox by default, otherwise it is just a button 😅

Reply:
Ah okay! That solves my comment to the above resolution, so I'll delete my comment. I missed this earlier. 
Thanks!